test/test_vmap.py

```diff
@@ -1365,6 +1365,31 @@ def test_expand_as(self):
         test(vmap(op), (torch.rand(B0, B1), torch.rand(B1, 2, 3, 5)), in_dims=(0, None))
         test(vmap(vmap(op)), (torch.rand(B0, B1, B2), torch.rand(B0, B1, B2, 2, 3, 5)))
 
+    def test_fill_and_zero_inplace(self):
+        test = functools.partial(self._vmap_test, check_propagates_grad=False)
+        B0, B1 = 7, 11
+        ops = (
+            lambda t: t.fill_(0.1),
+            lambda t: t.fill_(torch.tensor(0.2)),
+            lambda t: t.zero_(),
+        )
+
+        for op in ops:
+            # Single vmap, various in_dims / out_dims
+            test(op, [TensorFactory.randn([B0, 3])])
+            test(op, [TensorFactory.randn([2, 5, B0, 3])], in_dims=2)
+            test(op, [TensorFactory.randn([2, 5, B0, 3])], in_dims=2, out_dims=2)
+
+            # Doubly nested vmap
+            test(vmap(op), [TensorFactory.randn([B0, B1])])
+            test(vmap(op), [TensorFactory.randn([B1, 2, 5, B0, 3])], in_dims=2)
+            test(vmap(op, in_dims=2), [TensorFactory.randn([2, 5, B0, B1, 3])],
+                 in_dims=2, out_dims=2)
+
+        # test when value is a batched tensor for fill_ operator
+        B0, B1 = 3, 5
+        test(Tensor.fill_, [TensorFactory.randn([B0, B1]), TensorFactory.randn(B0)])
+
```

Comment:
Can you add a test for:
```
B0 = 3
a = torch.randn(10)
b = torch.randn(B0, 10)
vmap(Tensor.fill_, (None, 0))(a, b) ?
```
This should error out because `a` (the tensor being written to) isn't being vmapped over. I think this is what the implementation does as stated, but it would be good to check that it errors out in a test. We should also check that the error message is sane (with something like `self.assertRaisesRegex`.

Reply:
Thanks so much for the detailed suggestion, I have updated the PR with a test case for `not vmapped-over tensor`.

Reply:
The test code is updated by specifying `None` for the `in_dims` of `self` tensor, making it has no map dimension.
The `Runtime Error` regarding `broadcast shape` will be thrown, while the vmap function call is successful for the same inputs if we do not specify `None` for the `in_dims` parameter.